crates/api_common/src/custom_emoji.rs

```diff
@@ -0,0 +1,41 @@
+use crate::sensitive::Sensitive;
+use lemmy_db_views::structs::CustomEmojiView;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+pub struct CreateCustomEmoji {
+  pub category: String,
+  pub shortcode: String,
+  pub image_url: String,
+  pub alt_text: String,
+  pub keywords: String,
+  pub auth: Sensitive<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+pub struct EditCustomEmoji {
+  pub id: i32,
+  pub category: String,
+  pub shortcode: String,
+  pub image_url: String,
+  pub alt_text: String,
+  pub keywords: String,
+  pub auth: Sensitive<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+pub struct DeleteCustomEmoji {
+  pub id: i32,
+  pub auth: Sensitive<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct DeleteCustomEmojiResponse {
+  pub id: i32,
+  pub success: bool,
```

Comment:
Success should be expressed through status code. You probably just need to return `()` here, I dont think there is any need for the id either.

Reply:
The 'success' response I copied from elsewhere in the codebase, i forget where; Some other 'Delete' endpoint does it.

The id is actually for the frontend. Since WebSocket callbacks aren't doable, I need to know which entity to remove in the table. Technically, I guess I could remove it before the Delete is saved, but could be incorrect if it throws an error.

Reply:
The 'success' response I copied from elsewhere in the codebase, i forget where; Some other 'Delete' endpoint does it.

The id is actually for the frontend. Since WebSocket callbacks aren't doable, I need to know which entity to remove in the table. Technically, I guess I could remove it before the Delete is saved, but could be incorrect if it throws an error.

Reply:
Yes there are other success fields, feel free to return whatever extra info you need for websockets.